src/algebra/algebra/basic.lean

```diff
@@ -593,6 +593,14 @@ def mk' (f : A →+* B) (h : ∀ (c : R) x, f (c • x) = c • f x) : A →ₐ[
 
 @[simp] lemma coe_mk' (f : A →+* B) (h : ∀ (c : R) x, f (c • x) = c • f x) : ⇑(mk' f h) = f := rfl
 
+/-- If a `ring_hom` between `R`-algebras commutes with the `algebra_map`s,
+  then it is an `alg_hom`. -/
+def of_comp_eq (f : A →+* B) (h : f.comp (algebra_map R A) = algebra_map R B) : A →ₐ[R] B :=
```

Comment:
does `@[simps]` not do the right thing here?

Reply:
I think `@[simps]` only generates lemmas about projections, but coercion isn't syntactically the same (only defeq to) a projection? But probably I should use `@[simp]` or just make it an abbreviation.

Reply:
it's worth trying with `@[simps?]` and seeing what it says, it's definitely generated coe lemmas for me before (for example with some power basis stuff)